tests/foreman/cli/test_provisioning.py

```diff
@@ -15,15 +15,424 @@
+        to=settings.vlan_networking.dhcp_to,
+        mask=settings.vlan_networking.netmask,
+        dns_primary=settings.vlan_networking.dns_primary,
+        domain=[domain],
+        dns=capsule,
+        dhcp=capsule,
+        discovery=capsule,
+        # httpboot=capsule,   # TODO
+        template=capsule,
+        tftp=capsule,
+        remote_execution_proxy=[capsule],
+    ).create()
+    return subnet
+
+
+@pytest.fixture(scope='session')
+def lce(user_credentials, org):
+    return entities.LifecycleEnvironment(
+                user_credentials,
+                organization=org).create()
+
+
+@pytest.fixture(scope='session')
+def rhel7(user_credentials, org):
+    repo_id = enable_rhrepo_and_fetchid(
+        basearch='x86_64',
+        org_id=org.id,
+        product=PRDS['rhel'],
+        repo=REPOS['rhel7']['name'],
+        reposet=REPOSET['rhel7'],
+        releasever=REPOS['rhel7']['releasever'],
+    )
+    repo = entities.Repository(user_credentials, id=repo_id)
+    response = repo.sync(synchronous=False)
+    entities.ForemanTask(id=response['id']).poll(timeout=1800)
+    return repo.read()
+
+
+@pytest.fixture(scope='session')
+def rhel7ks(user_credentials, org):
+    repo_id = enable_rhrepo_and_fetchid(
+        basearch=REPOS['rhel7ks']['arch'],
+        org_id=org.id,
+        product=REPOS['rhel7ks']['product'],
+        releasever=REPOS['rhel7ks']['releasever'],
+        repo=REPOS['rhel7ks']['name'],
+        reposet=REPOS['rhel7ks']['reposet'],
+    )
+    repo = entities.Repository(user_credentials, id=repo_id)
+    response = repo.sync(synchronous=False)
+    entities.ForemanTask(id=response['id']).poll(timeout=1800)
+    return repo.read()
+
+
+@pytest.fixture(scope='session')
+def rhel8ks(user_credentials, org):
+    # First RHEL8 AppStream KS
+    repo_id = enable_rhrepo_and_fetchid(
+            basearch=REPOS['rhel8ksappstream']['arch'],
+            org_id=org.id,
+            product=REPOS['rhel8ksappstream']['product'],
+            releasever=REPOS['rhel8ksappstream']['releasever'],
+            repo=REPOS['rhel8ksappstream']['name'],
+            reposet=REPOS['rhel8ksappstream']['reposet'],
+    )
+    repo = entities.Repository(user_credentials, id=repo_id)
+    response = repo.sync(synchronous=False)
+    # Now RHEL8 BaseOS KS which we will return at the end
+    repo_id = enable_rhrepo_and_fetchid(
+            basearch=REPOS['rhel8ksbaseos']['arch'],
+            org_id=org.id,
+            product=REPOS['rhel8ksbaseos']['product'],
+            releasever=REPOS['rhel8ksbaseos']['releasever'],
+            repo=REPOS['rhel8ksbaseos']['name'],
+            reposet=REPOS['rhel8ksbaseos']['reposet'],
+    )
+    repo = entities.Repository(user_credentials, id=repo_id)
+    response = repo.sync(synchronous=False)
+    # Wait for sync to finish
+    entities.ForemanTask(id=response['id']).poll(timeout=1800)
+    entities.ForemanTask(id=response['id']).poll(timeout=100)
```

Comment:
why we are polling here 2 times?

Reply:
Oh. That was because we are syncing two repos and I was movig lines I think. Fixed.